src/scripts/main.js

```diff
@@ -1 +1,64 @@
 'use strict';
+
+const body = document.querySelector('body');
+
+function notification(text) {
+  body.insertAdjacentHTML('beforeend',
+    `<div class="warning" data-qa="notification">${text}</div>`);
+}
+
+const firstPromise = new Promise((resolve, reject) => {
+  function handleClick(ev) {
+    ev.preventDefault();
+
+    resolve('First promise was resolved!');
+  }
+
+  document.addEventListener('click', handleClick);
+
+  setTimeout(() => {
+    reject(new Error());
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve, reject) => {
+  function handleClick(ev) {
+    ev.preventDefault();
+
+    resolve('Second promise was resolved!');
+  }
+
+  document.addEventListener('click', handleClick);
+  document.addEventListener('contextmenu', handleClick);
+});
+
+const thirdPromice = new Promise((resolve, reject) => {
+  let click;
+  let contextmenu;
+
+  const text = 'Third promise was resolved!';
+
+  document.addEventListener('click', () => {
+    click = true;
+
+    if (click && contextmenu) {
+      resolve(text);
+    }
```

Comment:
Move it to a separate function. You repeat it twice

Reply:
> Move it to a separate function. You repeat it twice

fixed